### PR TITLE
MLR 323

### PR DESCRIPTION
**Pull request by cschroed-usgs**

### CODE REVIEW

File: src/main/java/gov/usgs/wma/mlrgateway/service/LegacyCruService.java [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 import gov.usgs.wma.mlrgateway.StepReport;
 import gov.usgs.wma.mlrgateway.client.LegacyCruClient;
 import gov.usgs.wma.mlrgateway.controller.BaseController;
+import gov.usgs.wma.mlrgateway.workflow.LegacyWorkflowService;
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.logging.Level;
```
[the remark this answers is not in the document]
Comment (cschroed-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyCruService.java]: Thanks for catching that. I pushed a commit removing all unused imports in the project.

File: src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java [hunk 2]
```diff
@@ -35,29 +38,50 @@ public LegacyValidatorService(LegacyCruService legacyCruService, LegacyValidator
...
+		} catch (Exception e) {
+			if(e instanceof FeignBadResponseWrapper) {
+				duplicateValidationStatus = ((FeignBadResponseWrapper)e).getStatus();
+				BaseController.addStepReport(new StepReport(VALIDATION_STEP, duplicateValidationStatus,  ((FeignBadResponseWrapper)e).getBody(), ml.get(LegacyWorkflowService.AGENCY_CODE), ml.get(LegacyWorkflowService.SITE_NUMBER)));
+			} else {
+				duplicateValidationStatus = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+				BaseController.addStepReport(new StepReport(VALIDATION_STEP, duplicateValidationStatus,  e.getMessage(), ml.get(LegacyWorkflowService.AGENCY_CODE), ml.get(LegacyWorkflowService.SITE_NUMBER)));
+			}
+		}
 		try {
```
[the remark this answers is not in the document]
Comment (cschroed-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Yep! You're reading it right! My understanding was that we wanted all validations to be evaluated regardless of if other validations failed. I've been wrong before though :smile: @kmschoep-usgs, should we be executing all validations or exiting early if one fails?

[the remark this answers is not in the document]
Comment (zmoore-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Yeah, I think I remember that decision now as well. 

My only question then would be whether we would want the "Validation Step" step report to show up in the output twice if both pieces were to fail? I think as it is right now if duplicate validation fails we'd get a "Validation Step" step report added to the output and then if other validation fails as well we'd get a second "Validation Step" step report added to the output?

[the remark this answers is not in the document]
Comment (kmschoep-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Yep, keep going with other validations even if this one fails.  And what you have for the step outputs is fine.  I'm up to my elbows in step reports anyway, so I can rearrange if it needs to be.  Merging this is going to be interesting..

[the remark this answers is not in the document]
Comment (cschroed-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Ok, glad we're on the same page about the independence of validation execution! Yep! It does potentially add two "Validation Step" step reports with different detail messages. Is that bad? 
Some potential alternatives:
A) two step reports with different step names instead
B) Just one step report
The error messages from the two validation methods are pretty different. If we go with B), how  we combine them? Should it continue to be somewhat machine-parseable?
I'm guessing Kathy's current work rejiggering the JSON will change `StepReport`s. Will these decisions matter after that work is merged?

[the remark this answers is not in the document]
Comment (cschroed-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Sorry, that's getting to be a lot of questions. Please feel free to IM me or chat in person.

[the remark this answers is not in the document]
Comment (cschroed-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Just saw Kathy's comment. Thanks!

[the remark this answers is not in the document]
Comment (zmoore-usgs) [reply src/main/java/gov/usgs/wma/mlrgateway/service/LegacyValidatorService.java]: Yeah, I was mostly thinking ahead to once this is merged with Kathy's work. I'm not sure if it's a bad thing either, just wanted to make sure I was following the logic correctly. I think this is good to merge and any actual changes could be made within Kathy's work

**Pull request merged**